pkg/devfile/validate/validate.go

```diff
@@ -5,37 +5,36 @@ import (
 
 	adaptersCommon "github.com/openshift/odo/pkg/devfile/adapters/common"
 	"github.com/openshift/odo/pkg/devfile/parser/data"
+	v200 "github.com/openshift/odo/pkg/devfile/parser/data/2.0.0"
 	"github.com/openshift/odo/pkg/devfile/parser/data/common"
+	"github.com/openshift/odo/pkg/devfile/validate/generic"
 	"github.com/openshift/odo/pkg/util"
-	"k8s.io/klog"
 
-	v200 "github.com/openshift/odo/pkg/devfile/parser/data/2.0.0"
+	"k8s.io/klog"
 )
 
 // ValidateDevfileData validates whether sections of devfile are odo compatible
 func ValidateDevfileData(data interface{}) error {
 	var components []common.DevfileComponent
 	var commandsMap map[string]common.DevfileCommand
-	var events common.DevfileEvents
+
+	// Validate the generic devfile data before validating odo specific logic
+	if err := generic.ValidateDevfileData(data); err != nil {
+		return err
+	}
 
 	switch d := data.(type) {
 	case *v200.Devfile200:
 		components = d.GetComponents()
 		commandsMap = d.GetCommands()
-		events = d.GetEvents()
 
 		// Validate all the devfile components before validating commands
 		if err := validateComponents(components); err != nil {
 			return err
 		}
 
 		// Validate all the devfile commands before validating events
-		if err := validateCommands(d.Commands, commandsMap, components); err != nil {
-			return err
-		}
-
-		// Validate all the events
-		if err := validateEvents(events, commandsMap, components); err != nil {
+		if err := validateCommands(commandsMap); err != nil {
```

Comment:
I think `ValidateContainerName`, should also be part of generic validation, it could be added inside component validation

Reply:
There are a few reasons why I did not do it:
1. validation pkg is generally for validating logic directly related to the devfile schema
2. having said point 1, this func should not have been in validate pkg in the first place and is relying on adapter util funcs which further adds reason behind my point no 1.

Odo is the only container based tool. Workspace may need it but I dont see Console using it? 🤔  
Lmk, what you think!

Reply:
@maysunfaisal It is ok to not to add in library, just wanted an opinion essentially it is the validation of component name specified in the devfile, the container component would be translated to containers to all other tools, no?  what console is translating devfile containers to if specified?

Reply:
@adisky I've updated the pkg to validate the component names and made it more cleaner, pls have a look! Thx.